tracers/lightstep/lightstep.go

```diff
@@ -3,43 +3,59 @@ package main
 import (
 	"errors"
 	"fmt"
+	"net"
 	"strconv"
 	"strings"
 
 	lightstep "github.com/lightstep/lightstep-tracer-go"
 	opentracing "github.com/opentracing/opentracing-go"
 )
 
+const (
+	defComponentName = "skipper"
+)
+
 func InitTracer(opts []string) (opentracing.Tracer, error) {
-	var token, host string
+	componentName := defComponentName
 	var port int
+	var host, token string
+
 	for _, o := range opts {
-		switch {
-		case strings.HasPrefix(o, "token="):
-			token = o[6:]
-		case strings.HasPrefix(o, "collector="):
-			parts := strings.Split(o[10:], ":")
-			host = parts[0]
-			if len(parts) == 1 {
-				port = 443
-			} else {
-				var err error
-				aport, err := strconv.Atoi(parts[1])
-				if err != nil {
-					return nil, fmt.Errorf("failed to parse %s as int: %s", parts[1], err)
-				}
-				port = int(aport)
+		parts := strings.SplitN(o, "=", 2)
+		switch parts[0] {
+		case "component-name":
+			if len(parts) > 1 {
+				componentName = parts[1]
+			}
+		case "token":
+			token = parts[1]
+		case "collector":
+			var err error
+			var sport string
+
+			host, sport, err = net.SplitHostPort(parts[1])
+			if err != nil {
+				return nil, err
 			}
-		}
 
+			port, err = strconv.Atoi(sport)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse %s as int: %s", sport, err)
```

Comment:
Just a nit, but please use %v in the format string for err, because errors.Error does not implement stringer interface.

Reply:
I dind't touch that neither, anyway I don't mind setting to `%v` if it's prefered.

Just to let more information regarding this. `fmt` knows how to print `Error` and `stringer` types: 
*https://github.com/golang/go/blob/555eb70db2f7ff4aa22768339425430c1dfdf6ac/src/fmt/print.go#L594-L604
* https://play.golang.org/p/JNNUM5Hgho6